components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/EntitlementPolicyAdminService.java

```diff
@@ -578,7 +579,7 @@ public PaginatedStatusHolder getStatusData(String about, String key, String type
         Set<PAPStatusDataHandler> handlers = EntitlementAdminEngine.getInstance().
                 getPapStatusDataHandlers();
         for (PAPStatusDataHandler handler : handlers) {
-            if (handler instanceof SimplePAPStatusDataHandler) {
+            if (DAOFactory.getPAPStatusDataHandler().getClass().isInstance(handler))  {
```

Comment:
What if we check for `handler instanceof SimplePAPStatusDataHandler` instead of exact class depending on the `DAOFactory`, would it break any funtionality?

Reply:
Ack